Debug tests insist on spike/openocd being in `/bin`

I'm using a clean install of Zorin OS Lite 16.2 (Ubuntu 20.04 LTS derivative) on a VirtualBox virtual machine with all available updates applied and nothing else installed other than prerequisites needed to build the `riscv-gnu-toolchain` tools (including Spike) and OpenOCD.

I tried to run the smoke test against Spike as per the instructions here:

* https://github.com/riscv-software-src/riscv-tests/tree/master/debug#usage

> To run a quick smoke test against spike, run `make`.

with the following tools available via `PATH`:
```
echo $PATH
/home/user/Downloads/riscv-gnu-toolchain/installed-tools/bin:/home/user/Downloads/riscv-openocd/installed-tools/bin:/usr/lib/ccache:/home/user/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

which riscv64-unknown-elf-gcc
/home/user/Downloads/riscv-gnu-toolchain/installed-tools/bin/riscv64-unknown-elf-gcc

which riscv64-unknown-elf-gdb
/home/user/Downloads/riscv-gnu-toolchain/installed-tools/bin/riscv64-unknown-elf-gdb

which spike
/home/user/Downloads/riscv-gnu-toolchain/installed-tools/bin/spike

which openocd
/home/user/Downloads/riscv-openocd/installed-tools/bin/openocd
```
but it failed as follows:
```
make
./gdbserver.py \
	./targets/RISC-V/spike64.py \
	CeaseMultiTest \
	--isolate \
	--print-failures \
	--sim_cmd /bin/spike \
	--server_cmd /bin/openocd \
	
Using $misa from hart definition: 0x8000000000141125
[CeaseMultiTest] Starting > logs/20230628-175631-spike64-CeaseMultiTest.log
[CeaseMultiTest] not_applicable in 0.00s
::::::::::::::::::::::::::::[ ran 1 tests in 0s ]:::::::::::::::::::::::::::::
1 tests returned not_applicable
./gdbserver.py \
	./targets/RISC-V/spike64.py \
	CeaseRunTest \
	--isolate \
	--print-failures \
	--sim_cmd /bin/spike \
	--server_cmd /bin/openocd \
	
Using $misa from hart definition: 0x8000000000141125
[CeaseRunTest] Starting > logs/20230628-175631-spike64-CeaseRunTest.log
[CeaseRunTest] not_applicable in 0.00s
::::::::::::::::::::::::::::[ ran 1 tests in 0s ]:::::::::::::::::::::::::::::
1 tests returned not_applicable
./gdbserver.py \
	./targets/RISC-V/spike64.py \
	CeaseStepiTest \
	--isolate \
	--print-failures \
	--sim_cmd /bin/spike \
	--server_cmd /bin/openocd \
	
Using $misa from hart definition: 0x8000000000141125
[CeaseStepiTest] Starting > logs/20230628-175632-spike64-CeaseStepiTest.log
[CeaseStepiTest] not_applicable in 0.00s
::::::::::::::::::::::::::::[ ran 1 tests in 0s ]:::::::::::::::::::::::::::::
1 tests returned not_applicable
./gdbserver.py \
	./targets/RISC-V/spike64.py \
	CheckMisa \
	--isolate \
	--print-failures \
	--sim_cmd /bin/spike \
	--server_cmd /bin/openocd \
	
Using $misa from hart definition: 0x8000000000141125
[CheckMisa] Starting > logs/20230628-175632-spike64-CheckMisa.log
[CheckMisa] exception in 0.21s
Test: CheckMisa
Target: spike64
---------------------------------[ Compile ]----------------------------------
+ riscv64-unknown-elf-gcc -g programs/checksum.c programs/tiny-malloc.c programs/infinite_loop.S -DDEFINE_MALLOC -DDEFINE_FREE programs/entry.S programs/init.c -DNHARTS=1 -I ../env -T targets/RISC-V/spike64.lds -nostartfiles -mcmodel=medany -DXLEN=64 -o /tmp/spike64_checksum-8000000000141125_bw1bi42t -march=rv64imafc -mabi=lp64
--------------------------------[ Traceback ]---------------------------------
Traceback (most recent call last):
  File "/home/user/Downloads/riscv-tests/debug/testlib.py", line 1203, in run
    self.classSetup()
  File "/home/user/Downloads/riscv-tests/debug/testlib.py", line 1257, in classSetup
    BaseTest.classSetup(self)
  File "/home/user/Downloads/riscv-tests/debug/testlib.py", line 1166, in classSetup
    self.target_process = self.target.create()
  File "/home/user/Downloads/riscv-tests/debug/./targets/RISC-V/spike64.py", line 23, in create
    return testlib.Spike(self, isa="RV64IMAFC", progbufsize=0,
  File "/home/user/Downloads/riscv-tests/debug/testlib.py", line 101, in __init__
    self.process = subprocess.Popen(cmd, stdin=subprocess.PIPE,
  File "/usr/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/bin/spike'
-------------------------------[ End of logs ]--------------------------------
Result: exception
Logfile: logs/20230628-175632-spike64-CheckMisa.log
Reproduce: ./gdbserver.py ./targets/RISC-V/spike64.py CheckMisa
Time elapsed: 0.21s
::::::::::::::::::::::::::::[ ran 1 tests in 0s ]:::::::::::::::::::::::::::::
1 tests returned exception
   CheckMisa > logs/20230628-175632-spike64-CheckMisa.log
make: *** [Makefile:20: run.spike64.CheckMisa] Error 1
```
I copied the Spike executable to `/bin` but then it failed again:
```
sudo cp `which spike` /bin
`
make
./gdbserver.py \
	./targets/RISC-V/spike64.py \
	CeaseMultiTest \
	--isolate \
	--print-failures \
	--sim_cmd /bin/spike \
	--server_cmd /bin/openocd \
	
Using $misa from hart definition: 0x8000000000141125
[CeaseMultiTest] Starting > logs/20230628-180529-spike64-CeaseMultiTest.log
[CeaseMultiTest] not_applicable in 0.00s
::::::::::::::::::::::::::::[ ran 1 tests in 0s ]:::::::::::::::::::::::::::::
1 tests returned not_applicable
./gdbserver.py \
	./targets/RISC-V/spike64.py \
	CeaseRunTest \
	--isolate \
	--print-failures \
	--sim_cmd /bin/spike \
	--server_cmd /bin/openocd \
	
Using $misa from hart definition: 0x8000000000141125
[CeaseRunTest] Starting > logs/20230628-180530-spike64-CeaseRunTest.log
[CeaseRunTest] not_applicable in 0.00s
::::::::::::::::::::::::::::[ ran 1 tests in 0s ]:::::::::::::::::::::::::::::
1 tests returned not_applicable
./gdbserver.py \
	./targets/RISC-V/spike64.py \
	CeaseStepiTest \
	--isolate \
	--print-failures \
	--sim_cmd /bin/spike \
	--server_cmd /bin/openocd \
	
Using $misa from hart definition: 0x8000000000141125
[CeaseStepiTest] Starting > logs/20230628-180530-spike64-CeaseStepiTest.log
[CeaseStepiTest] not_applicable in 0.00s
::::::::::::::::::::::::::::[ ran 1 tests in 0s ]:::::::::::::::::::::::::::::
1 tests returned not_applicable
./gdbserver.py \
	./targets/RISC-V/spike64.py \
	CheckMisa \
	--isolate \
	--print-failures \
	--sim_cmd /bin/spike \
	--server_cmd /bin/openocd \
	
Using $misa from hart definition: 0x8000000000141125
[CheckMisa] Starting > logs/20230628-180530-spike64-CheckMisa.log
[CheckMisa] exception in 0.33s
Test: CheckMisa
Target: spike64
---------------------------------[ Compile ]----------------------------------
+ riscv64-unknown-elf-gcc -g programs/checksum.c programs/tiny-malloc.c programs/infinite_loop.S -DDEFINE_MALLOC -DDEFINE_FREE programs/entry.S programs/init.c -DNHARTS=1 -I ../env -T targets/RISC-V/spike64.lds -nostartfiles -mcmodel=medany -DXLEN=64 -o /tmp/spike64_checksum-8000000000141125__k8pc8qc -march=rv64imafc -mabi=lp64
-------------------------[ /tmp/spike-_ryv6nh2.log ]--------------------------
+ /bin/spike -p1 --isa RV64IMAFC --dm-auth --dm-progsize 0 --dm-sba 64 --dm-abstract-rti 30 -m0x1212340000:0x10000000 --rbb-port 0 /tmp/spike64_checksum-8000000000141125__k8pc8qc
Listening for remote bitbang connection on port 39067.
warning: tohost and fromhost symbols not in ELF; can't communicate with target

--------------------------------[ Traceback ]---------------------------------
Traceback (most recent call last):
  File "/home/user/Downloads/riscv-tests/debug/testlib.py", line 1203, in run
    self.classSetup()
  File "/home/user/Downloads/riscv-tests/debug/testlib.py", line 1257, in classSetup
    BaseTest.classSetup(self)
  File "/home/user/Downloads/riscv-tests/debug/testlib.py", line 1170, in classSetup
    self.server = self.target.server(self)
  File "/home/user/Downloads/riscv-tests/debug/targets.py", line 170, in server
    return testlib.Openocd(server_cmd=self.server_cmd,
  File "/home/user/Downloads/riscv-tests/debug/testlib.py", line 367, in __init__
    self.process = self.start(cmd, logfile, extra_env)
  File "/home/user/Downloads/riscv-tests/debug/testlib.py", line 372, in start
    process = subprocess.Popen(cmd, stdin=subprocess.PIPE,
  File "/usr/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/bin/openocd'
-------------------------[ /tmp/spike-_ryv6nh2.log ]--------------------------
+ /bin/spike -p1 --isa RV64IMAFC --dm-auth --dm-progsize 0 --dm-sba 64 --dm-abstract-rti 30 -m0x1212340000:0x10000000 --rbb-port 0 /tmp/spike64_checksum-8000000000141125__k8pc8qc
Listening for remote bitbang connection on port 39067.
warning: tohost and fromhost symbols not in ELF; can't communicate with target

-------------------------------[ End of logs ]--------------------------------
Result: exception
Logfile: logs/20230628-180530-spike64-CheckMisa.log
Reproduce: ./gdbserver.py ./targets/RISC-V/spike64.py CheckMisa
Time elapsed: 0.33s
::::::::::::::::::::::::::::[ ran 1 tests in 0s ]:::::::::::::::::::::::::::::
1 tests returned exception
   CheckMisa > logs/20230628-180530-spike64-CheckMisa.log
make: *** [Makefile:20: run.spike64.CheckMisa] Error 1
```
I then copied the OpenOCD executable to `/bin` but again it failed as per this log:

[20230628-171707-spike64-EtriggerTest.log](https://github.com/riscv-software-src/riscv-tests/files/11898248/20230628-171707-spike64-EtriggerTest.log)

--- a/debug/Makefile
+++ b/debug/Makefile
@@ -1,4 +1,3 @@
-RISCV_SIM ?= spike
 XLEN ?= 64
 
 src_dir ?= .
@@ -22,8 +21,6 @@ run.%:
 		$(word 3, $(subst ., ,$@)) \
 		--isolate \
 		--print-failures \
-		--sim_cmd $(RISCV)/bin/$(RISCV_SIM) \
-		--server_cmd $(RISCV)/bin/openocd \
 		$(if $(EXCLUDE_TESTS),--exclude-tests $(EXCLUDE_TESTS))
 
 # Target to check all the multicore options.

--- a/debug/README.md
+++ b/debug/README.md
@@ -9,11 +9,10 @@ confident that the actual debug interface is functioning correctly.
 Requirements
 ============
 The following should be in the user's path:
-* riscv64-unknown-elf-gcc (`rvv-0.9.x` branch for riscv-gnu-toolchain should
-  work if master does not have vector support yet)
-* riscv64-unknown-elf-gdb (can be overridden with `--gdb` when running
-  gdbserver.py manually), which should be the latest from
-  git://sourceware.org/git/binutils-gdb.git.
+* riscv64-unknown-elf-gcc (GCC 12 and later should work). If your binary has a
+  different name, you can set the RISCV_TESTS_DEBUG_GCC environment variable.
+* riscv64-unknown-elf-gdb. If your binary has a
+  different name, you can set the RISCV_TESTS_DEBUG_GDB environment variable.
 * spike (can be overridden with `--sim_cmd` when running gdbserver.py
   manually), which should be the latest from
   https://github.com/riscv/riscv-isa-sim.git.

